### PR TITLE
[Windows] Add render method information to the debug OSD

### PR DESCRIPTION
**Pull request by CrystalP**
## Description

Added line to the debug OSD for Windows render method information.

For now, name and optional render method provided string.
With PR#23359 the intermediate target format will be added.

The DXVA render method returns the status of the Video Super Resolution feature.

## Motivation and context

The render method changes depending on dxva support for input/output conversions and it's convenient to have the information on screen rather than in debug log only.
Same for the new dxva super video resolution scalers activation status.

## How has this been tested?

Windows and auto/dxva/pixel shaders/software renderer render method settings, including situations with fallback from dxva to ps.

Unable to test super resolution, no suitable hardware on hand.

## What is the effect on users?

Helpful for devs.

## Screenshots (if appropriate):
image

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by thexai** [comment]
Looks good:

osd

File: xbmc/cores/VideoPlayer/VideoRenderers/DebugInfo.h
Comment (thexai) [inline xbmc/cores/VideoPlayer/VideoRenderers/DebugInfo.h:27]: Maybe rename to "render"

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.h
Comment (thexai) [inline xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.h:192]: Probably don't worth initialize (empty string is fine)

File: xbmc/cores/VideoPlayer/VideoRenderers/DebugInfo.h
Comment (CrystalP) [reply xbmc/cores/VideoPlayer/VideoRenderers/DebugInfo.h:27]: sure

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.h
Comment (CrystalP) [reply xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.h]: sure

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp
Comment (thexai) [inline xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp:308]: Sorry, now thinking that as final decision on Super Resolution enabled or not is in Nvidia driver and driver settings (Kodi only request enable) and is not possible confirm if is enabled or not (no API available yet) is better change this to:

`requested enable`

to avoid users confusion

**Review by thexai — APPROVED** [review]

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp
Comment (CrystalP) [reply xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp:308]: that's fine, I didn't use ON for that reason. Just "requested" is enough I think.

**Comment by fuzzard** [comment]
Jenkins build this please

**Comment by thexai** [comment]
Jenkins build this please

**Comment by CrystalP** [comment]
@KarellenX Wiki update is not possible at this moment (problem with screenshot upload).
Want me to wait for the Wiki to be fixed or could I merge and update the wiki when it's available?

**Comment by KarellenX** [comment]
Yes, best to merge. No need to hold this up for the wiki.
Thanks.

**Comment by KarellenX** [comment]
I've made a report here... https://github.com/xbmc/infrastructure/issues/48

**Comment by KarellenX** [comment]
>I didn't figure out how to change the "Page updated for v20" at the top to v21.

At the bottom of the wiki page, you will see... `{{updated|20}}` 
Simply change to `{{updated|21}}`
I've already done it on that page.

**Pull request merged**
